### ISSUE TITLE
False positive OperatorSpacing for JS function chaining

### ISSUE THREAD

**Issue by spl1nes** [PHPCSStandards/PHP_CodeSniffer#495]
## Describe the bug

This is not an important issue for me since it's just a little extra thing, but I do sometimes use this CodeSniffer for JS as well. 

In JS (`?.`) we can do function chaining with null checks, just like in php (`?->`). When I use `Squiz.WhiteSpace.OperatorSpacing` in php the sniffer correctly recognizes that `?->` is "one" operator and doesn't expect white spaces around `?`. For consistency it would be good if it would also recognize `?.` as "one" operator in JS and NOT demand a white space around `?`.

### Code sample

```js
const adventurer = {
    name: 'Alice',
    cat: {
        name: 'Dinah',
    },
  };

  const dogName = adventurer.dog?.name;
  console.log(dogName);
  // Expected output: undefined

  console.log(adventurer.someNonExistentMethod?.());
  // Expected output: undefined
```

### Custom ruleset
```xml
<?xml version="1.0"?>
<ruleset name="My Custom Standard">
<rule ref="Squiz.WhiteSpace.OperatorSpacing">
    <properties>
        <property name="ignoreNewlines" value="true"/>
    </properties>
</rule>
</ruleset>
```

### To reproduce

```sh
--------------------------------------------------------------------------------------------------------
FOUND 4 ERRORS AFFECTING 2 LINES
--------------------------------------------------------------------------------------------------------
  8 | ERROR | [x] Expected 1 space before "?"; 0 found (Squiz.WhiteSpace.OperatorSpacing.NoSpaceBefore)
  8 | ERROR | [x] Expected 1 space after "?"; 0 found (Squiz.WhiteSpace.OperatorSpacing.NoSpaceAfter)
 12 | ERROR | [x] Expected 1 space before "?"; 0 found (Squiz.WhiteSpace.OperatorSpacing.NoSpaceBefore)
 12 | ERROR | [x] Expected 1 space after "?"; 0 found (Squiz.WhiteSpace.OperatorSpacing.NoSpaceAfter)
--------------------------------------------------------------------------------------------------------
PHPCBF CAN FIX THE 4 MARKED SNIFF VIOLATIONS AUTOMATICALLY
--------------------------------------------------------------------------------------------------------
```

## Expected behavior

No errors.

## Versions (please complete the following information)

|                                       |                                                                                                           |
|--------------------------|----------------------------------------------------------------------  |
| Operating System           | linux |
| PHP version                    | 8.2   |
| PHP_CodeSniffer version | 3.9.2 |
| Standard                         | Squiz                                          |
| Install type                      | Composer local |

## Please confirm

- [x] I have searched the issue list and am not opening a duplicate issue.
- [x] I confirm that this bug is a bug in PHP_CodeSniffer and not in one of the external standards.
- [x] I have verified the issue still exists in the `master` branch of PHP_CodeSniffer.

**Comment by spl1nes** [comment]
Oh, nvm then. Didn't know about the dropped support for JS.
